File: ObjectiveC/TwitterPage/Common/Network/TwitterRequestParser.m
//
//  TwitterRequestParser.m
//  TwitterPage
//
//  Created by DMITRY SINYOV on 30.08.17.
//  Copyright © 2017 DMITRY SINYOV. All rights reserved.
//

#import "TwitterRequestParser.h"
#import "TweetModel.h"

@implementation TwitterRequestParser

+(NSArray <TweetModel*>*)parseTimeline:(id)JSON {
    if (![JSON isKindOfClass:[NSArray class]]) return nil;
    
    NSMutableArray <TweetModel*>* tweets = [NSMutableArray array];
    for (NSDictionary *tweetDict in JSON)
    {
        TweetModel *tweet = [[TweetModel alloc] initWithJSON:tweetDict];
        if (tweet && !tweet.isIncorrect) [tweets addObject:tweet];
    }
    return tweets;
}

@end

File: SWIFT/TwitterPageSWIFT/Common/Network/TwitterRequestParser.swift
//
//  TwitterRequestParser.swift
//  TwitterPageSWIFT
//
//  Created by DMITRY SINYOV on 16.09.17.
//  Copyright © 2017 DMITRY SINYOV. All rights reserved.
//

import UIKit

class TwitterRequestParser: NSObject {
    class func parseTimeline(json: Any?) -> [TweetModel]? {
        if (json == nil) { return nil }
        if (!(json is [[String: AnyObject]]?)) { return nil }
        var tweets = [TweetModel]()
        for tweetDict in json as! [[String: AnyObject]] {
            let tweet = TweetModel(json: tweetDict)
            if (!tweet.isIncorrect) { tweets.append(tweet) }
        }
        return tweets
    }
}
